woocommerce/woocommerce-ios
# [Woo POS] Present full-screen views when item list is empty or returns an error

Closes: #13360
Closes: #12815
Closes: #13480


## Description
Previously, if an error happened when loading products, we displayed it within the `ItemList` view, but the cart and the rest of the components could also be seen on screen. 

This PR changes this to show a full-screen error view instead, both for the cases when the item list fails by throwing an error, or when the item list loads with zero products. 

In each case we display a different view. The UI for both screens still needs some adjustments and tweaking to fit the latest designs, but this can be tackled separately.

## Testing
- Load POS
- Intercept the call to`ItemListViewModel.populatePointOfSaleItems()` and throw an error
- Observe that the error view is shown.
- Tap the Retry button
- Do not intercept it, observe that the dashboard is loaded normally.
- Repeat by changing `ItemListViewModel.populatePointOfSaleItems()` to return an empty list of items
```diff
- items = try await itemProvider.providePointOfSaleItems()
+ items = []
```
- Observe the empty view is shown.

| Error view | Empty view |
|--------|--------|
| ![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2024-08-09 at 11 32 10](https://github.com/user-attachments/assets/27011b86-f6dc-4110-b238-92115934c941) | ![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2024-08-09 at 11 33 21](https://github.com/user-attachments/assets/5b2b6d0c-42f6-481a-a4f9-398f3f71a41d) |
